controller/internal/supervisor/iptablesctrl/iptables.go

```diff
@@ -561,10 +561,11 @@ type ipsetInfo struct {
 }
 
 type aclIPset struct {
-	ipset     string
-	ports     []string
-	protocols []string
-	policy    *policy.FlowPolicy
+	ipset      string
+	ports      []string
+	protocols  []string
+	extensions map[string][]string
+	policy     *policy.FlowPolicy
```

Comment:
see above. I would just make this a []string .. and pass them to iptables .. we could keep it as a map with the idea that we will type and customize some of them in the future. So, there are "raw" entries that we just pass along and there can be typed entries that we know how to handle.

Reply:
Why do you want to have two fields when we can do this in one ?. We can have the map key as `raw` and pass the all its values blindly to the iptables, so the code looks something like this with both functionalities

```
		if byteCodes, ok := rule.extensions["bpf"]; ok {
		for _, byteCode := range byteCodes {
			rulesspec := []string{
				"-p", proto,
				"--match", "multiport", "--dports", strings.Join(rule.ports, ","),
				"-m", "bpf",
				"--bytecode", byteCode,
				"-j", "DROP",
			}
			if err := i.ipt.Append(i.appPacketIPTableContext, chain, rulesspec...); err != nil {
				return fmt.Errorf("unable to program rule for extension bpf: %v", err)
			}
		}
	}

	if rules, ok := rule.extensions["raw"]; ok {
		if err := i.ipt.Append(i.appPacketIPTableContext, chain, rules...); err != nil {
			return fmt.Errorf("unable to program rule for extension bpf: %v", err)
		}
	}
	
```